src/middlewared/middlewared/plugins/kubernetes_linux/lifecycle.py

```diff
@@ -129,11 +130,17 @@ async def status_change_internal(self):
     async def setup_pool(self):
         config = await self.middleware.call('kubernetes.config')
         await self.create_update_k8s_datasets(config['dataset'])
+        # Now we would like to setup catalogs
+        await self.middleware.call('catalog.sync_all')
 
     @private
     async def create_update_k8s_datasets(self, k8s_ds):
         for dataset in await self.kubernetes_datasets(k8s_ds):
             if not await self.middleware.call('pool.dataset.query', [['id', '=', dataset]]):
+                if os.path.exists(os.path.join('/mnt', dataset)):
+                    await self.middleware.run_in_thread(
+                        lambda: shutil.rmtree(os.path.join('/mnt', dataset), ignore_errors=True)
```

Comment:
Why do we have to use rmtree?

Its a very dangerous function we should avoid if possible